server/src/word-list.ts

```diff
@@ -0,0 +1,127 @@
+import {
+    TextDocument,
+    CompletionItem,
+    CompletionItemKind,
+    TextDocumentChangeEvent
+} from 'vscode-languageserver';
+
+// A data structure to keep track of the words used in this file
+// We currently use a set, can be changed to a trie for more efficient lookup
+class WordListClass {
+
+    // The keywords 
+    keywords : Set<string>;
+
+    // A mapping from files to the words in that file
+    dictionary: Map<TextDocument, Set<string>>;
+
+    /**
+     * Initialize the list to only contain the keywords
+     *
+     * @param {string[]} word
+     * 
+     */
+
+    constructor(word: string[]) {
```

Comment:
Should be called `words`

Reply:
maybe should be called "keywords", cause that's what field you are populating with it.

Reply:
Would it be better to have the constructor not rely on the keywords being passed in, but rather directly get them from `lex.ts`?

Reply:
That tightens coupling of the two classes which doesn't seem like a good idea to me

Reply:
That tightens coupling of the two classes which doesn't seem like a good idea to me

Reply:
Makes sense!

Reply:
I'm a little confused, towards the end of the file we have a keywords list that gets placed into our class set of keywords. Right now I just copied the keywords from lex.ts and placed that in the keywords list. You suggested to just import the list of keywords from lex.ts. If I do so, then our constructor won't need to rely on the keywords being passed in, but instead can just get it from lex.ts like angelaz said?

Reply:
The class should still take in a list of of keywords, but instead of copying and pasting them, it would be better to get the from the lexer to avoid code duplication.

Reply:
Do we need to take in a list of keywords? We could instead just get the keywords from the lexer in getList() and insert into the returning list?

Reply:
That creates additional coupling between the lexer and this wordlist class which I think is unnecessary

Reply:
How is this different than what you're suggesting (getting the keywords from the lexer and putting it into the keywords set in the constructor)?

Reply:
What I'm suggesting means that the WordList doesn't need to know anything about the lexer. Instead this responsibility is moved to server.ts. This could come in handy later if we want to enforce C0/C1 (which have different sets of keywords)

Reply:
Ohh ok that makes sense, thank you!